src/HTSPDemuxer.cpp

```diff
@@ -191,6 +208,15 @@ void CHTSPDemuxer::Speed ( int speed )
   SendSpeed();
 }
 
+void CHTSPDemuxer::Weight ( uint32_t weight )
+{
+  CLockObject lock(m_conn.Mutex());
```

Comment:
I'd move this lock to `CHTSPDemuxer::SendWeight` to keep the lock scope as small as possible (that's how we do it in most other places).